corehq/util/django_migrations.py

```diff
@@ -110,3 +113,40 @@ def noop_migration():
     Old migrations can be simply removed. `operations = []` is perfectly valid
     """
     return RunPython(RunPython.noop, RunPython.noop)
+
+
+def run_management_command(command_name, *args, required_commit=None, **kwargs):
+    """Return a migration operation that can be used to run a management command from
+    a Django migration. This will give the user directions for running the command
+    manually if the command fails or has since been removed from the codebase"""
+    @skip_on_fresh_install
+    def _run_command(apps, schema_editor):
+        safely_run_management_command(command_name, required_commit, *args, **kwargs)
+
+    return migrations.RunPython(_run_command, reverse_code=migrations.RunPython.noop, elidable=True)
+
+
+def safely_run_management_command(command_name, *args, required_commit=None, custom_message=None, **kwargs):
+    """Helper function to run a management command automatically and abort if there was an error
+    with helpful console output to allow users to run it manually if it fails or
+    has since been removed from the codebase."""
+    try:
+        call_command(command_name, *args, **kwargs)
+    except Exception:
+        traceback.print_exc()
+        print(f"""
+            A migration must be performed before this environment can be upgraded to the latest version
+            of CommCareHQ. This migration is run using the management command {command_name}.
+        """)
+        if required_commit or custom_message:
+            print("")
+            print(custom_message or f"""
+            Run the following commands to run the migration and get up to date:
```

Comment:
These commands are geared toward someone running HQ with commcare-cloud. I assume developers are expected to know how to adapt them to their dev environment context. Is that right?

Reply:
I don't think dev's every really hit this since they either keep things up to date or are doing fresh setups in which case this mostly won't get triggered because they are once off's that don't need to be run on fresh installs.

Reply:
I've hit these before when trying to bring my local HQ, which I rarely run, up to date.

Reply:
The commit is listed in the message so it should be relatively straight forward to figure out. Otherwise I could add a flag to check `settings.DEBUG`. What do you think?